src/python/TaskWorker/MasterWorker.py

```diff
@@ -392,6 +392,42 @@ def _lockWork(self, limit, getstatus, setstatus):
 
         return True
 
+    def runCanary(self, limit):
+        # Decide whether to use canary_name based on the canary_fraction value.
+        # canary_fraction is a float value in the range [0.0-1.0]
+        use_canary = random.random() < float(self.config.TaskWorker.canary_fraction)
```

Comment:
you can add these lines here
```python
if not use_canary:
   return True
```
avoid the `if` at line 402, avoid calling the REST to rewrite TWNAME with same value (!) and avoid the `if` at line 427

Reply:
Excellent suggestion, thanks!